File: mobile/src/main/kotlin/de/scout/fireplace/firebase/FirebaseModule.java
package de.scout.fireplace.firebase;

import dagger.Binds;
import dagger.Module;
import de.scout.fireplace.Configuration;
import de.scout.fireplace.Reporting;
import javax.inject.Singleton;

@Module(includes = FirebaseConfigurationModule.class)
public abstract class FirebaseModule {

  @Binds
  @Singleton
  abstract Configuration config(FirebaseConfiguration firebaseConfig);

  @Binds
  abstract Reporting reporting(FirebaseReporting reporting);
}

File: mobile/src/main/kotlin/de/scout/fireplace/firebase/FirebaseModule.kt
package de.scout.fireplace.firebase

import dagger.Binds
import dagger.Module
import de.scout.fireplace.Configuration
import de.scout.fireplace.Reporting
import de.scout.fireplace.inject.ApplicationScope

@Module(includes = arrayOf(FirebaseConfigurationModule::class))
internal interface FirebaseModule {

  @Binds
  @ApplicationScope
  fun config(firebaseConfig: FirebaseConfiguration): Configuration

  @Binds
  fun reporting(reporting: FirebaseReporting): Reporting
}
